conversation/frontend/src/routes/pages/folder-display.tsx

```diff
@@ -1,18 +1,239 @@
-import { Outlet } from 'react-router-dom';
+import { LoaderFunctionArgs, useNavigate } from 'react-router-dom';
+import { MessageMetadata } from '~/models';
+import {
+  folderQueryOptions,
+  useFolderMessages,
+  useMarkRead,
+  useMarkUnread,
+} from '~/services';
+import { MessagePreview } from './message-preview';
+import {
+  IconAdd,
+  IconEdit,
+  IconFilter,
+  IconReadMail,
+  IconUnreadMail,
+} from '@edifice.io/react/icons';
+import {
+  Button,
+  Dropdown,
+  EmptyScreen,
+  List,
+  Loading,
+  SearchBar,
+  ToolbarItem,
+  useBreakpoint,
+  useEdificeTheme,
+} from '@edifice.io/react';
+import clsx from 'clsx';
+import { useEffect, useState } from 'react';
+import {
+  useAppActions,
+  useFilterUnreadMessageList,
+  useSelectedMessageIds,
+} from '~/store/actions';
+import { useTranslation } from 'react-i18next';
 import { useSelectedFolder } from '~/hooks';
+import illuMessagerie from '@images/emptyscreen/illu-messagerie.svg';
 
-export const loader = (_queryClient: QueryClient) => async () => {
-  return null;
-};
+export const loader =
+  (_queryClient: QueryClient) =>
+  async ({ params }: LoaderFunctionArgs) => {
+    const messagesQuery = folderQueryOptions.getMessages(params.folderId!);
+    const messages = await _queryClient.ensureQueryData(messagesQuery);
+    return { messages };
+  };
 
 export function Component() {
   const { folderId } = useSelectedFolder();
+  const { theme } = useEdificeTheme();
+  const navigate = useNavigate();
+
+  const { t } = useTranslation('conversation');
+
+  const [searchText, setSearchText] = useState('');
+  const {
+    setSearchMessageList,
+    setSelectedMessageIds,
+    setFilterUnreadMessageList,
+  } = useAppActions();
+  const selectedIds = useSelectedMessageIds();
+  const filterUnread = useFilterUnreadMessageList();
+  const markAsReadQuery = useMarkRead();
+  const markAsUnreadQuery = useMarkUnread();
+  const { md } = useBreakpoint();
+  const filterEnum = {
+    unread: 'UNREAD',
+  };
+
+  const {
+    messages,
+    isPending: isLoadingMessage,
+    isFetchingNextPage: isLoadingNextPage,
+    hasNextPage,
+    fetchNextPage,
+  } = useFolderMessages(folderId!);
+
+  const handlerChangeSearchText = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const newText = event.target.value;
+    setSearchText(newText);
+
+    if (newText === '') {
+      setSearchMessageList(newText);
+    }
+  };
+
+  const handleClickSearch = () => {
+    setSearchMessageList(searchText);
+  };
+
+  const handleMarkAsRead = () => {
+    markAsReadQuery.mutate({ id: selectedIds });
+  };
+
+  const handleMarkAsUnread = () => {
+    markAsUnreadQuery.mutate({ id: selectedIds });
+  };
+
+  const toolbar: ToolbarItem[] = [
+    {
+      type: 'button',
+      name: 'read',
+      props: {
+        children: (
+          <>
+            <IconReadMail />
+            <span>{t('tag.read')}</span>
+          </>
+        ),
+        onClick: () => {
```

Comment:
```suggestion
        onClick: handleMarkAsRead
```
Evite des lambdas inutiles = bon pour la RAM et la planète